src/inputs/BasicInputs/BasicDateInput/BasicDateInput.js

```diff
@@ -43,10 +55,19 @@ export const BasicDateInput = (props) => {
             inputFormat={format}
             minDate={dateProps.minDate}
             maxDate={dateProps.maxDate}
-            renderInput={(params) => <TextField variant="outlined" sx={{ flexGrow: 1 }} size="small" {...params} />}
+            renderInput={(params) => (
+              <TextField
+                variant="outlined"
+                sx={{ flexGrow: 1 }}
+                size="small"
+                error={!!errorType}
+                helperText={errorMessages[errorType] ?? ''}
+                {...params}
+              />
+            )}
             id={`date-input-${id}`}
             onChange={changeSelectedDate}
-            value={value}
+            value={value ?? new Date(undefined)}
```

Comment:
Do you realy want to create an invalid date here ?

Reply:
It's a rather dirty trick to keep error state on the input because `TextField` doesn't display error state for not required fields when value is `null`. I tried to force it by many ways but the only thing that worked was to replace `null` by something invalid (thus `InvalidDate` returned by `new Date(undefined)`)